### PR TITLE
♻️  Remove `version` from legacy mode object

### PR DESCRIPTION
**Pull request by rcebulko**
Part of #32693

### CODE REVIEW

File: src/mode-object.js [hunk 1]
```diff
@@ -29,7 +29,6 @@ export function getModeObject(opt_win) {
     esm: IS_ESM,
     test: getMode(opt_win).test,
     log: getMode(opt_win).log,
-    version: getMode(opt_win).version,
```
Comment (jridgewell) [inline src/mode-object.js:32]: Is `mode-object` use anywhere? We might need to update this to use the core method, so that the object stays the same.

Comment (rcebulko) [reply src/mode-object.js:32]: Yes, one place (`src/iframe-attributes.js`) in the process of updating and fixing the other test fail

**Review by jridgewell — APPROVED** [review]
Yah, `getModeObject` is used to store the return value, which can then be `JSON.stringify`'ed or some-such. It'll be a bit tricker to remove that usage, but at least it's very isolated.

**Comment by rcebulko** [comment]
> Yah, `getModeObject` is used to store the return value, which can then be `JSON.stringify`'ed or some-such. It'll be a bit tricker to remove that usage, but at least it's very isolated.

@jridgewell I just did a scan, and nothing references that stringified `version`; it's only used to initialize the value for `getMode` itself, so I'm removing it entirely including from the `ModeDef` type

**Review by samouri — APPROVED** [review]

**Pull request merged**